docs/documentation.md

```diff
@@ -27,112 +28,251 @@ To make editing changes to [Guides and examples](https://coremltools.readme.io/)
+	```shell
+	git add --all
+	git commit -m "Commit message for gh-pages"
+	git push origin gh-pages
+	```
+    
+	At this point you can share the following URL to preview the new version (substituting your user name for `sample-user`): `https://sample-user.github.io/coremltools/`.
+
+6. On your fork, switch to the `gh-pages` branch, and request a Pull Request (PR) for your changes to merge them with the `gh-pages` branch on the Apple repo. 
 
+Once approved, these changes will eventually be merged from your `gh-pages` branch to the Apple repo's `gh-pages` branch. 
 
-## Generating the top-level RST files
 
-This section describes how to generate a new version of the [API Reference](https://apple.github.io/coremltools/index.html) from scratch. It also shows how to edit the Sphinx navigation and content selection elements.
+## Generating a new API reference
 
+This section is for generating a new version of the API Reference from scratch, and also shows how to edit the Sphinx navigation and content selection elements.
 
-### Set up your environment and Sphinx
-​
-You need to do this only once. Follow these steps:
 
-1. In the `coremltools` root directory, create a Conda environment and install Sphinx, numpydoc, and the theme: 
+### Set up the environment
+
+1. [Fork and clone](#fork-and-clone) the coremltools repo.
+
+2. [Install Sphinx and theme](#install-sphinx-and-theme) in the root directory (`coremltools`) if you have not already done so.
+
+3. From the `main` branch, do a build for coremltools:
    
-   ```shell
-   conda create coremltools-env
-   conda activate coremltools-env
-   conda install sphinx
-   conda install numpydoc
-   conda install sphinx_rtd_theme
-   ```
+	```shell
+	zsh ./scripts/build.sh
+	```
+	
+	*Warning*: Include the `./` or the script may not function correctly.
+	
+	The build files appear in the `build` folder.
 
-2. Create a new `docs` folder in the root folder:
+    If you run into a `cmake` error (something about `cmake` not found in your PATH), you may need to change your PATH. If you need help with this, see [Use environment variables in Terminal on Mac](https://support.apple.com/guide/terminal/use-environment-variables-apd382cc5fa-4f58-4449-b20a-41c53c006f8f/mac). You may also need to install cmake.
+    
+    For example, you may want to switch to the folder where cmake is supposed to be:
+   
+    ```shell
+    (base) tonybove@users-MacBook-Pro coremltools % cd ~/Library/Python/3.8/bin
+    ```
     
+    Then follow these steps:
+    
+       1. Install cmake:
+           
+	      ```
+	      (base) tonybove@users-MacBook-Pro bin % pip install cmake
+          ...
+		  WARNING: The scripts cmake, cpack and ctest are installed in
+		  '/Users/tonybove/Library/Python/3.9/bin' which is not on PATH.
+		  Consider adding this directory to PATH...
+	      ```
+       
+       2. Put `/Users/tonybove/Library/Python/3.9/bin` on your PATH:
```

Comment:
This instruction doesn't really make sense for other people, since your username is in the path.

Reply:
Removed. Thanks!